### ISSUE TITLE
Test pollution: TestUpdateStream.test_set_insert_id can fail due to TestUpdateStream.test_service_switch

### ISSUE THREAD

**Issue by aaijazi** [youtube/vitess#136]
test_set_insert_id can fail if it's running on a slow disk (i.e., not SSD/ramdisk). This is bad because it's part of our `make site_test` suite, and presumably should pass even in less resource-heavy environments.

The particular failure mode that I saw was:

<pre>
-- 2014-10-29 19:50:43,369 update_stream:185 DEBUG Test Service Disabled: Pass
-- 2014-10-29 19:50:43,371 update_stream:202 DEBUG _test_service_enabled starting @ {'MariaDB': '0-62344-10'}
-- 2014-10-29 19:50:43,371 utils:182 DEBUG run: ['mariadb/bin/vtctl', '-log_dir', 'mariadb/vtdataroot/tmp', '-topo_implementation', 'zook
eeper', '-tablet_manager_protocol', 'bson', '-tablet_protocol', 'gorpc', 'ChangeSlaveType', 'test_nj-0000062345', 'replica'] 
-- 2014-10-29 19:50:43,849 update_stream:204 DEBUG sleeping a bit for the replica action to complete
-- 2014-10-29 19:50:53,855 update_stream:218 DEBUG Test Service Enabled: Pass
-- 2014-10-29 19:51:23,858 update_stream:233 DEBUG Testing enable -> disable switch starting @ {'MariaDB': '0-62344-10'}
-- 2014-10-29 19:51:23,892 utils:182 DEBUG run: ['mariadb/bin/vtctl', '-log_dir', 'mariadb/vtdataroot/tmp', '-topo_implementation', 'zook
eeper', '-tablet_manager_protocol', 'bson', '-tablet_protocol', 'gorpc', 'ChangeSlaveType', 'test_nj-0000062345', 'spare'] 
-- 2014-10-29 19:51:24,280 update_stream:247 ERROR Test Service Switch: FAIL
-- 2014-10-29 19:51:24,280 utils:182 DEBUG run: ['mariadb/bin/vtctl', '-log_dir', 'mariadb/vtdataroot/tmp', '-topo_implementation', 'zook
eeper', '-tablet_manager_protocol', 'bson', '-tablet_protocol', 'gorpc', 'ChangeSlaveType', 'test_nj-0000062345', 'replica'] 
-- 2014-10-29 19:51:24,671 utils:50 INFO ===== test_service_switch (__main__.TestUpdateStream) ... ok
-- 2014-10-29 19:51:24,671 utils:50 INFO ===== test_set_insert_id (__main__.TestUpdateStream)
-- 2014-10-29 19:51:24,671 utils:50 INFO ===== test_set_insert_id (__main__.TestUpdateStream) ... 
-- 2014-10-29 19:51:24,897 update_stream:339 DEBUG test_set_insert_id: starting @ {'MariaDB': '0-62344-428'}
-- 2014-10-29 19:51:24,982 utils:50 INFO ===== test_set_insert_id (__main__.TestUpdateStream) ... FAIL
-- 2014-10-29 19:51:24,982 update_stream:126 DEBUG Tearing down the servers and setup
-- 2014-10-29 19:51:24,982 tablet:624 DEBUG killing vttablet: test_nj-0000062344
-- 2014-10-29 19:51:24,982 tablet:624 DEBUG killing vttablet: test_nj-0000062345
-- 2014-10-29 19:51:24,983 vtclient:127 WARNING db connection failed: test_keyspace.0.master <host-redacted>:15009, ('retry: Query server is in NOT_SERVING state', 'SqlQuery.GetSessionId', '<Ta
bletConnection <host-redacted>:15009 master test_keyspace/0>')
-- 2014-10-29 19:51:24,983 vtdb_logger:64 WARNING vtclient_exception for test_keyspace.0.master: ('unable to create vt connection', 'test_keyspace.0.master', '<host-redacted>:15009', RetryError
(('retry: Query server is in NOT_SERVING state', 'SqlQuery.GetSessionId', '<TabletConnection <host-redacted>:15009 master test_keyspace/0>'),))
Exception in thread write_thd:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "test/update_stream.py", line 197, in perform_writes
    self._exec_vt_txn(self._populate_vt_insert_test)
  File "test/update_stream.py", line 269, in _exec_vt_txn
    vtdb_conn = self._vtdb_conn('localhost:%u' % master_tablet.port)
  File "test/update_stream.py", line 263, in _vtdb_conn
    conn.connect()
  File "mariadb/py-vtdb/vtdb/vtclient.py", line 95, in connect
    return self._connect()
  File "mariadb/py-vtdb/vtdb/vtclient.py", line 133, in _connect
    'unable to create vt connection', db_key, host_addr, db_exception)
OperationalError: ('unable to create vt connection', 'test_keyspace.0.master', '<host-redacted>:15009', RetryError(('retry: Query server is in NOT_SERVING state', 'SqlQuery.GetSessionId', '<Tab
letConnection <host-redacted>:15009 master test_keyspace/0>'),))
</pre>

Basically, it looks like the previously run test_service_switch put the cluster into a "NOT_SERVING" state as part of what it's testing. It tries to put the cluster back in an enabled state, but doesn't actually wait for state to fully change. This creates a race condition which can cause subsequent tests to fail.

We probably want to add something like a wait_step condition at the end of the test, and block until the state is what we expect it to be.
